src/features/navigate/Navigate.tsx

```diff
@@ -0,0 +1,103 @@
+import './Navigate.css';
+import img from '../../assets/logo.png';
+import {
+  AppBar,
+  Box,
+  Button,
+  Tab,
+  Tabs,
+  Toolbar,
+  Typography,
+} from '@mui/material';
+import { useEffect, useState } from 'react';
+import { Outlet, useNavigate } from 'react-router-dom';
+import { useAccount, useDisconnect } from 'wagmi';
+import { useWeb3Modal } from '@web3modal/react';
+import {
+  initDataProtector,
+  selectAppIsConnected,
+  selectThereIsSomeRequestPending,
+} from '../../app/appSlice';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { shortAddress } from '../../utils/utils';
+
+export default function Navigate() {
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+  const { open } = useWeb3Modal();
+  const { address, isConnected, isDisconnected } = useAccount();
+  const { disconnect } = useDisconnect();
+  const [value, setValue] = useState('myProtectedData');
+
+  //get the state from the store
+  const isAccountConnected = useAppSelector(selectAppIsConnected);
+  const loading = useAppSelector(selectThereIsSomeRequestPending);
+
+  const handleChange = (event: React.SyntheticEvent, newValue: string) => {
+    setValue(newValue);
+  };
+
+  useEffect(() => {
+    if (isDisconnected) {
+      open();
+    } else if (isConnected && !isAccountConnected) {
+      dispatch(initDataProtector());
+    }
+  }, [isDisconnected, isAccountConnected]);
+
+  useEffect(() => {
+    if (value === 'myProtectedData') {
+      navigate('/protectedData');
+    }
+    if (value === 'sendMail') {
+      navigate('/sendMail');
+    }
+  }, [value]);
```

Comment:
React Hook useEffect has a missing dependency: 'navigate'. Either include it or remove the dependency array. eslint [react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)